### PR TITLE
test: add Spark 4.2 parity scenarios and pin the divergences they expose

### PR DESCRIPTION
**Pull request by davidlghellin**
Adds 5444 executable scenarios to the BDD suite (6128 -> 11572), 2392 of them tagged @sail-bug. Every expected value was measured against Spark JVM 4.2.0, so a @sail-bug means Spark passes and Sail does not; it never encodes a guess.

Sources:
- Scenarios that sat in local branches and never reached main, deduplicated. Most were written next to a fix that never landed, so they fail on main.
- Targeted hunts over vector_cosine_similarity / vector_inner_product, the range of suffixed numeric literals, and the datetime functions (constructors and epoch conversions, date/timestamp arithmetic, intervals, parsing under ANSI, extraction and time zones, and the TIME type). Time-zone-dependent behaviour is also covered outside UTC (America/Los_Angeles across DST, Asia/Kolkata, Pacific/Chatham, Pacific/Pago_Pago); the harness pins the session to UTC, which hid every zone bug below.

Notable divergences pinned here:
- timestamp_seconds/millis/micros ignore the session time zone.
- TIMESTAMP + INTERVAL '1' DAY adds 24 h instead of one calendar day across DST; DATE - INTERVAL '12' HOUR drops the hours and returns a DATE.
- to_timestamp(<number>) reads seconds as microseconds.
- The minimum of each integral type cannot be written as a literal (-128Y, -32768S, -9223372036854775808L); -2147483648 is typed BIGINT.
- CAST(INTERVAL '2' HOUR AS INT) is 7200 instead of 2.

Suite hygiene found along the way:
- 21 scenarios were never collected because two outlines in the same file shared a name and pytest-bdd kept only the last one; 7 of them were already on main in csv/csv_options.feature.
- Error assertions that accepted any message containing ARRAY or FLOAT now assert Spark's error class.
- Scenarios whose expectation was wrong against the JVM were corrected, not dropped, including a few whose name stated the opposite of what Spark does.

Ports the shared step definitions the imported scenarios need, and excludes @sail-only scenarios, which the JVM cannot validate.

### CODE REVIEW

File: python/pysail/tests/spark/function/features/string/regexp_instr.feature [hunk 1]
```diff
@@ -38,3 +38,178 @@ Feature: regexp_instr output schema
...
+      Then query result
+        | result |
+        | 0      |
+
+  Rule: The idx argument does not affect the returned position
+
+    # idx is the "matched group id" but Spark always returns the start of the
+    # whole match, even when the requested group starts elsewhere or does not
+    # exist. These cases are exactly where the naive mapping to DataFusion's
+    # regexp_instr (3rd arg = search start) diverges.
```
[the remark this answers is not in the document]
Comment (davidlghellin) [reply python/pysail/tests/spark/function/features/string/regexp_instr.feature:76]: Thanks for the review, but this doesn't match Spark 4.2. `RegExpInStr` returns the start of the
whole match and never uses `idx` for the position, in both the interpreted and codegen paths
(`sql/catalyst/.../regexpExpressions.scala`, v4.2.0):

    if (m.find) {
      m.toMatchResult.start() + 1   // start() with no group index
    } else {
      0
    }

It also doesn't validate the group index: unlike `RegExpExtract`, its `nullSafeEval` never calls
`checkGroupIndex`, and it catches `IllegalStateException` and returns 0. So `idx = 2` and `idx = 5`
both return the whole-match start, which is exactly what these scenarios pin.

These scenarios were validated against Spark JVM 4.2.0: the full feature suite passes there with
0 failures, including the untagged ones in this rule. The behaviour described here is
`regexp_extract`'s, not `regexp_instr`'s.

**Comment by linhr** [comment]
Thanks for this exploration. I feel this provides valuable information to guide bug fixes. 

Let's keep this PR as a reference without merging this, since the large number of tests will likely slow down CI a lot. Let's think about a good way to follow up with this to have the issues addressed. We likely need to design some way to automate the fix and tests due to the number of edge cases discovered here.
